### DISCUSSION TITLE
Inline Grid Column/Row Definitions

### DISCUSSION THREAD

**Discussion by WeCarryBags** [unoplatform/uno#8969] [Q&A] [answered]
## What would you like to be added:

In WinUI 3 at least, it is possible to define inline Grid Column/Row definitions, as below (from the WinUI 3 Gallery app):

```
<Grid Width="240" Height="120" Background="Gray" ColumnDefinitions="50, Auto, *" RowDefinitions ="50, Auto, *">
    <Rectangle Fill="Red" Grid.Column="2" Grid.Row="2" />
    <Rectangle Fill="Blue" Grid.Row="1" />
    <Rectangle Fill="Green" Grid.Column="1" />
    <Rectangle Fill="Yellow" Grid.Row="1" Grid.Column="1" />
</Grid>
```

However, trying this in a hobby project results in the definitions being ignored.

## Why is this needed:

It provides compatibility with the latest XAML syntax available for WinUI 3, and it's cleaner than the current way.

```
<Grid>
<Grid.RowDefinisions>
<RowDefinition />
<RowDefinition />
</Grid.RowDefinisions>
</Grid>
```

## For which Platform:

I think a syntax change like this would apply to basically everything?

## Anything else we need to know?

I tried testing the Uno .NET 6 template in VS, which is where it didn't work. I might have an old version, but I don't know.

**Comment by jeromelaban** [comment, marked as answer]
This feature [has been implemented](https://platform.uno/docs/articles/features/windows-ui-succinct-syntax.html?q=succin) in Uno 4.3. 

Which version of Uno are you using, and on which platform are you testing it?

**Reply by WeCarryBags** [reply]
Thanks for the answer. I can confirm that the more concise markup works after creating a new Uno project and copying the XAML in there.

I'm not sure how to check the version of Uno that my project uses or how to upgrade, unfortunately.

Edit: I had a look at the package references in my project (new to this) and saw Uno 4.2.6 in use. I'll have a look at upgrading those references to latest using this guide: https://platform.uno/docs/articles/migrating-from-previous-releases.html .

Thanks again!

Second edit: I managed to upgrade the Uno packages successfully through the NuGet Package Manager in VS, but ran into some strange errors on the way before getting it working.

After upgrading through the NuGet Package Manager, my MainPage.xaml file was giving a few errors related to the Column/RowDefinitions already there. It said I couldn't change the value of a readonly property.

I tried cleaning the project, which didn't help, and eventually removed the lines containing inline Column/RowDefinitions to get the project to build again. After a successful build, I put the definition lines back again and suddenly it started working.

Just leaving these notes here in case someone with a similar issue finds it on Google, so that it may be of help.
